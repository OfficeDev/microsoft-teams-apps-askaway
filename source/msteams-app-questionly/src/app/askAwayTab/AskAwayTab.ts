import { PreventIframe } from 'express-msteams-host';

/**
 * Used as place holder for the decorators
 */
@PreventIframe('/askAwayTab/index.html')
@PreventIframe('/askAwayTab/config.html')
@PreventIframe('/askAwayTab/remove.html')
@PreventIframe('/askAwayTab/popups/createsession.html')
@PreventIframe('/askAwayTab/popups/qnaSessioncreatednotification.html')
export class AskAwayTab {}
