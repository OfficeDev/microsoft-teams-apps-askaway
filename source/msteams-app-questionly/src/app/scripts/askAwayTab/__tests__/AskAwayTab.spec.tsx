/**
 * @jest-environment jsdom
 */
// tslint:disable:no-relative-imports
import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { AskAwayTab } from '../AskAwayTab';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import TabContent from '../TabContent';
import MeetingPanel from '../MeetingPanel';

configure({ adapter: new enzymeAdapterReact16() });

describe('AskAwayTab Component', () => {
    it('should match the snapshot', () => {
        const wrapper = shallow(<AskAwayTab />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the MeetingPanel when frameContext is set to sidepanel', () => {
        const component = shallow(<AskAwayTab />);
        component.setState({
            frameContext: 'sidePanel',
        });

        expect(component.find(MeetingPanel)).toHaveLength(1);
    });

    it('should render the TabContent when frameContext is set to content', () => {
        const component = shallow(<AskAwayTab />);
        component.setState({
            frameContext: 'content',
        });

        expect(component.find(TabContent)).toHaveLength(1);
    });
});
