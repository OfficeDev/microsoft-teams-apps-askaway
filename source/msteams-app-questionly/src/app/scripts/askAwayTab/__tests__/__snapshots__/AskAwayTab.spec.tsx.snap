// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskAwayTab Component should match the snapshot 1`] = `
<Provider
  rtl={false}
  style={
    Object {
      "background": "unset",
    }
  }
  theme={Object {}}
>
  <div>
    <h1 />
  </div>
</Provider>
`;
